File: 2019_cpp/src/main/cpp/commands/LightRainbow.cpp
#include "commands/LightRainbow.h"

#include <cmath>
#include "Robot.h"

LightRainbow::LightRainbow() :
	m_r(0.0),
	m_g(0.0),
	m_b(0.0)
{
	Requires(&Robot::m_lights);
}

void LightRainbow::Initialize()
{
}

void LightRainbow::Execute()
{
	double h = (1.0 + std::cos(0.6 * TimeSinceInitialized())) / 2.0;
//	Color color = Color.getHSBColor(360.0f * h, 1.0f, 1.0f);
		
//	m_r = color.getRed() / 255.0;
//	m_b = color.getGreen() / 255.0;
//	m_g = color.getBlue() / 255.0;
	Robot::m_lights.SetColour(m_r, m_g, m_b);
}

bool LightRainbow::IsFinished()
{ 
	return false; 
}

void LightRainbow::End()
{
	Robot::m_lights.SetColour(0.0, 0.0, 0.0);
}

void LightRainbow::Interrupted()
{
	End();
}

File: 2019/src/main/java/frc/robot/commands/LightRainbow.java
package frc.robot.commands;

import java.awt.Color;

import edu.wpi.first.wpilibj.command.Command;
import frc.robot.Robot;

public class LightRainbow extends Command {
	private double m_r, m_g, m_b;

	public LightRainbow() {
		requires(Robot.m_lights);
	}
	
	@Override
	protected void initialize() {
	}

	@Override
	protected void execute() {
		float h = (1.0f + (float) Math.cos(0.6 * timeSinceInitialized())) / 2.0f;
		Color color = Color.getHSBColor(h, 1.0f, 1.0f);
		
		m_r = (double) color.getRed() / 255.0;
		m_b = (double) color.getGreen() / 255.0;
		m_g = (double) color.getBlue() / 255.0;
		Robot.m_lights.setColour(m_r, m_g, m_b);
	}

	@Override
	protected boolean isFinished() {
		return false;
	}

	@Override
	protected void end() {
		Robot.m_lights.setColour(0.0, 0.0, 0.0);
	}

	@Override
	protected void interrupted() {
		end();
	}
}
